src/windows/Program.cs

```diff
@@ -49,13 +52,15 @@ public static void RealTimeProcessing(string processName, double? minDurationFor
             }
         }
 
+        private const string LineSeparator = "--------------------------------------------------------------------------------------------------------";
+
         public static void RealTimeProcessing(int pid, double? minDurationForGCPausesInMSec)
         {
             Console.WriteLine();
             Process process = Process.GetProcessById(pid);
             Console.WriteLine($"Monitoring process with name: {process.ProcessName} and pid: {pid}");
-            Console.WriteLine("GC#{0,10} | {1,15} | {2,5} | {3,10}", "index", "type", "gen", "pause (ms)");
-            Console.WriteLine("----------------------------------------------------");
+            Console.WriteLine("GC#{0,10} | {1,15} | {2,5} | {3,10} | {4,10} | {5,10} | {6, 21} |", "index", "type", "gen", "pause (ms)", "promoted", "gen size", "reason");
```

Comment:
for "gen size", if it just did a gen1 GC, this is showing only the gen1 size, not gen0/2/3/4, even though it collected both gen0 and gen1. do you think users would want to see only gen1 size in this case? it seems like they might want to see the size of all the generations that got collected. an alternative is to show the whole heap size which is exposed as the `HeapSizeAfterMB` property (GCStats uses it [here](https://github.com/microsoft/perfview/blob/main/src/PerfView/GcStats.cs#L294)).

for promoted - most people would want to see total promoted in that GC, not limited to the generation (so if you are doing a gen1 GC, you'd want to see promotion from both gen0 and gen1 since both gens get collected). this is actually the sum of all the TotalPromotedSize* you have below, it's just that when a gen isn't getting collected its TotalPromotedSize would be 0 (eg, if you are doing a gen1 GC, TotalPromoted for gen2/3/4 would be 0). you can get this as the `PromotedMB` property like [here](https://github.com/microsoft/perfview/blob/main/src/PerfView/GcStats.cs#L296).

also it might be better to have the unit on this line instead of printing it on each line?

Reply:
Ah agreed! I wasn't sure the best way to show this without more columns, it seems like you'd want to see prev, collected, promoted, and new gen size for each collection action (per gen), I wonder how to best represent that - thoughts? Will push quick fixes for the comments below but agree this needs more thought!